examples/modules/example_module.cpp

```diff
@@ -105,10 +102,25 @@ int main(int argc, char** argv) {
         });
 
     Registry::register_resource(rr);
-    my_mod->add_model_from_registry(generic, m);
 
-    my_mod->start();
-    Server::start();
-    Server::wait();
+    // The `ModuleService_` must outlive the Server, so the declaration order
+    // here matters.
+    auto my_mod = std::make_shared<ModuleService_>(argv[1]);
+    Server server;
+
+    my_mod->add_model_from_registry(&server, generic, m);
+    my_mod->start(&server);
+
+    std::thread signal_wait_thread([&server, &sigset]() {
```

Comment:
You could also make a helper function to create and return this thread somehow.

Reply:
See my comment above.